packages/frontend/app/(dashboard)/course/[cid]/queue/[qid]/page.tsx

```diff
@@ -683,77 +683,119 @@ export default function QueuePage({ params }: QueuePageProps): ReactElement {
+                                    .catch((e) => {
+                                      const errorMessage = getErrorMessage(e)
+                                      message.error(errorMessage)
+                                      throw e
+                                    })
                                 })
+                              Promise.all(promises)
                                 .catch((e) => {
-                                  const errorMessage = getErrorMessage(e)
-                                  message.error(errorMessage)
-                                  throw e
+                                  message.error(
+                                    'One or more status updates failed:' + e,
+                                  )
                                 })
-                            },
-                          )
-                          Promise.all(promises)
-                            .catch((e) => {
-                              message.error(
-                                'One or more status updates failed:' + e,
-                              )
-                            })
-                            .finally(() => {
-                              setIsFinishAllHelpingButtonLoading(false)
-                            })
-                        }}
+                                .finally(() => {
+                                  setIsFinishAllHelpingButtonLoading(false)
+                                })
+                            }}
+                          />
+                        </span>
+                      </Tooltip>
+                    )}
+                  </div>
+                  {helpingQuestions
+                    .filter((q) => q.status !== OpenQuestionStatus.Paused)
+                    .map((question: Question) => {
+                      return (
+                        <QuestionCard
+                          key={question.id}
+                          question={question}
+                          cid={cid}
+                          qid={qid}
+                          configTasks={configTasks}
+                          studentAssignmentProgress={studentAssignmentProgress}
+                          isStaff={isStaff}
+                          isBeingHelped={true}
+                        />
+                      )
+                    })}
+                  <Divider className={'my-4'} />
+                </>
+              )}
+              {helpingQuestions.filter(
+                (q) => q.status == OpenQuestionStatus.Paused,
+              ).length > 0 && (
+                <>
+                  <div className="flex items-center justify-between">
+                    <QueueHeader
+                      text="Paused Questions"
+                      visibleOnDesktopOrMobile="both"
+                    />
+                  </div>
+                  {helpingQuestions
+                    .filter((q) => q.status === OpenQuestionStatus.Paused)
```

Comment:
I guess these filters is kinda why I don't think paused questions should be considered as being helped. Could we perhaps memoize this or something so we have a `helpingQuestions` and a `pausedQuestions` variables?